typing/printtyp.ml

```diff
@@ -1670,7 +1670,11 @@ let group_syntactic_items x =
         remove_row (s_elt :: acc) rem
     | Sig_class_type _ as src :: tydecl1 :: tydecl2 :: rem ->
         remove_row ({src; ghosts = ghosts @ [tydecl1; tydecl2]}::acc) rem
-    | src :: rem -> remove_row ({src; ghosts} :: acc) rem
+    | [Sig_class _] | [Sig_class _; _ ] | [Sig_class _; _;_ ] -> assert false
+    | [Sig_class_type _] | [Sig_class_type _; _ ] -> assert false
```

Comment:
Have you tried the version where there is a single `Sig_class` case, with a nested match on the rest of the list, which contains the error cases? It might be nicer (it makes it easier to understand what are the asserted-false situations).

Reply:
I don't find the change that convincing but why not.